src/finalizationManager.ts

```diff
@@ -36,39 +39,8 @@ export class FinalizationManager {
     this.finalizeAttempts = config.get<number>('externalClientsConfig.clientsUrls.jobManager.finalizeTasksAttempts');
   }
 
-  public async sendExportCallbacks(
-    job: JobExportResponse | JobFinalizeResponse,
-    callbackParams: ICallbackDataExportBase | ICallbackExportResponse
-  ): Promise<void> {
-    try {
-      this.logger.info({ jobId: job.id, callbacks: job.parameters.callbacks, msg: `Sending callback for job: ${job.id}` });
-      const targetCallbacks = job.parameters.callbacks;
-      if (!targetCallbacks) {
-        return;
-      }
-      const callbackPromises: Promise<void>[] = [];
-      for (const target of targetCallbacks) {
-        const params: ICallbackExportData = { ...callbackParams, roi: job.parameters.roi };
-        callbackPromises.push(this.callbackClient.send(target.url, params));
-      }
-
-      const promisesResponse = await Promise.allSettled(callbackPromises);
-      promisesResponse.forEach((response, index) => {
-        if (response.status === 'rejected') {
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-          this.logger.error({ reason: response.reason, url: targetCallbacks[index].url, jobId: job.id, msg: `Failed to send callback to url` });
-        }
-      });
-    } catch (error) {
-      this.logger.error({ err: error, callbacksUrls: job.parameters.callbacks, jobId: job.id, msg: `Sending callback has failed` });
-    }
-  }
-
-  public async jobFinalizePoll(): Promise<boolean> {
-    const finalizeTask = await this.queueClient.queueHandlerForFinalizeTasks.dequeue<ITaskFinalizeParameters>(this.finalizeTaskType);
-    if (!finalizeTask) {
-      return false;
-    }
+  @withSpanAsyncV4
+  private async runFinalize(finalizeTask: ITaskResponse<ITaskFinalizeParameters>): Promise<boolean> {
```

Comment:
use span context from job params as is done in cache-seeder

Reply:
as with cache-seeder you have saved the trace context inside the task / job params
please use that context when creating the span as was done in cache-seeder